### PR TITLE
Azure Monitor: Bug Fix for incorrect variable cascading for template variables

### LINKED ISSUE #47477: Azure Monitor variables not cascading properly
**What happened**:
Right now, when cascading variables, it was failing to update to the value `None` when going from a combination that produced results to one that did not have any results: 
cascadingVariablesNotUpdatingh

**What you expected to happen**: Variables should update to None if there are no results returned: 

variableProperlyCascading

**How to reproduce it (as minimally and precisely as possible)**: create the following variables: 
- sub: Subscriptions()
- rg: ResourceGroups($sub)
- ns: Namespaces($sub, $rg)
- res: ResourceNames($sub, $rg, $ns)

Go back to the dashboard and play around with the dropdowns until you eventually switch from a variable that returns something for namespaces or resources to one that does not.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.5
- Data source type & version: Azure Monitor Datasource
- OS Grafana is installed on: Windows
- User OS & Browser: Edge, Windows
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by jcolladokuri**
**What this PR does / why we need it**: Adds a minor fix so that variables cascade properly: 

variableProperlyCascading

**Which issue(s) this PR fixes**: Right now, when cascading variables, it was failing to update to the value `None` when going from a combination that produced results to one that did not have any results: 
cascadingVariablesNotUpdatingh

Fixes #47477 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export class VariableSupport extends CustomVariableSupport<DataSource, AzureMoni
         try {
           const templateVariablesResults = await this.callGrafanaTemplateVariableFn(queryObj.grafanaTemplateVariableFn);
           return {
-            data: templateVariablesResults ? [toDataFrame(templateVariablesResults)] : [],
+            data: (templateVariablesResults && templateVariablesResults.length > 0) ? [toDataFrame(templateVariablesResults)] : [],
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts:35]: nit:

```suggestion
            data: templateVariablesResults?.length ? [toDataFrame(templateVariablesResults)] : [],
```

**Review by andresmgot — COMMENTED** [review]
Thanks for sending the patch!

Can you add a test for this? I believe it can be something like:

```ts
    it('should return empty data', (done) => {
      const variableSupport = new VariableSupport(
        createMockDatasource({
          azureLogAnalyticsDatasource: {
            defaultSubscriptionId: 'defaultSubscriptionId',
          },
          getMetricNames: jest.fn().mockResolvedValueOnce([]),
        })
      );
      const mockRequest = {
        targets: [
          {
            refId: 'A',
            queryType: AzureQueryType.GrafanaTemplateVariableFn,
            grafanaTemplateVariableFn: {
              kind: 'MetricNamesQuery',
              rawQuery: 'metricNames(resourceGroup, metricDefinition, resourceName, metricNamespace)',
            },
          } as AzureMonitorQuery,
        ],
      } as DataQueryRequest<AzureMonitorQuery>;
      const observables = variableSupport.query(mockRequest);
      observables.subscribe((result: DataQueryResponseData) => {
        expect(result.data.length).toBe(0);
        done();
      });
    });
```

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts (hunk 1 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts]: Thanks for the suggestions, I have added the test and incorporated this

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.test.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import createMockDatasource from './__mocks__/datasource';
 import { invalidSubscriptionError } from './__mocks__/errors';
 import { AzureMonitorQuery, AzureQueryType } from './types';
 import { VariableSupport } from './variables';
+import { it } from 'test/lib/common';
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.test.ts:8]: can you remove this line? It's causing the lint errors here: https://drone.grafana.net/grafana/grafana/57546/1/9 (`it` is automagically imported when using `jest`)

**Review by andresmgot — COMMENTED** [review]
LGTM! You just need to remove one import line.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.test.ts (hunk 2 above)
Comment (jcolladokuri) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.test.ts]: Done!

**Review by sarahzinger — APPROVED** [review]
🎉

**Pull request merged**